src/cursor/cur_std.c

```diff
@@ -1327,6 +1332,9 @@ __wt_cursor_bounds_restore(
            bounds_state->upper_bound.size)) != 0)
         WT_RET_PANIC(session, ret, "Unrecoverable error encountered while restoring bounds");
 
+    __wt_free(session, bounds_state->lower_bound);
+    __wt_free(session, bounds_state->upper_bound);
```

Comment:
The free needs to happen after the colgroup loop.

Reply:
And we should use `__wt_buf_free` to match the allocator (`__wt_buf_set`).

Reply:
Ah yes you're right, I put this in the wrong place. Thanks! Will also change it to `__wt_buf_free`